x-pack/test/common/utils/server_route_repository/create_supertest_service_from_repository.ts

```diff
@@ -98,7 +98,10 @@ export function getApiClientFromSupertest<TServerRouteRepository extends ServerR
       const { method, pathname, version } = formatRequest(endpoint, params.path);
       const url = format({ pathname, query: params?.query });
 
-      const headers: Record<string, string> = { 'kbn-xsrf': 'foo' };
+      const headers: Record<string, string> = {
+        'kbn-xsrf': 'foo',
+        'x-elastic-internal-origin': 'kibana',
```

Comment:
Had to add that in a central spot so we can also test internal APIs

Reply:
Both `kbn-xsrf` and `x-elastic-internal-origin` will be assigned when you pass `withInternalHeaders: true,` in 

```
const supertestAdmin = await st.getSupertestWithRoleScope('admin', {
    useCookieHeader: true,
    withInternalHeaders: true,
  });
```